lib/compat/wordpress-6.6/rest-api.php

```diff
@@ -29,3 +29,23 @@ function wp_api_template_access_controller( $args, $post_type ) {
 	}
 }
 add_filter( 'register_post_type_args', 'wp_api_template_access_controller', 10, 2 );
+
+function gutenberg_block_editor_preload_paths_6_6( $paths ) {
+	$paths[] = array( '/wp/v2/settings', 'OPTIONS' );
+	$paths[] = '/wp/v2/settings';
+	$paths[] = array( '/wp/v2/templates', 'OPTIONS' );
+	$paths[] = '/wp/v2/types?context=edit';
+	$paths[] = '/?_fields=' . urlencode( implode( ',', [
```

Comment:
Do we really need to make the request using the "fields". If the request is not cached, the fields make the request faster maybe at time but it has downsides:

 - We need to cache using the fields
 - We may have to do two requests (if we need other fields)

So I think we should really think twice before considering "fields" argument. The perf impact need to be huge :)

Reply:
The `__unstableBase` entity definition includes fields in `baseURLParams`, so we need to change that as well.

Reply:
It looks like I skipped the `route` fields and only fetch public site settings. See #39256.

Reply:
@jsnajdr any thoughts

Reply:
These fields never change, as far as I understand, so why is it not ok? Why we are making these unstable base requests is another question though.

Reply:
> Why we are making these unstable base requests is another question though.

Non-admin users can't access site settings, so some of the required settings were made available via index endpoint.

Reply:
Why can't they access settings? (in context=view)

Reply:
I think it doesn’t support the view context.

Reply:
`_fields` must be specified explicitly because otherwise the `/` REST endpoint returns a monstrous `routes` field that includes detailed metadata about all registered REST endpoints. On my local WordPress with Woo and few other plugins installed, this field has 1.2MB.

I optimized this back in #39256, where I also argued for not preloading `/` at all, because the data there are non-essential. But that was for Post Editor, maybe Site Editor is different.

Reply:
Makes sense, I guess @ellatrix is probably right, settings should have been context view in the settings endpoint instead :)

Reply:
So I don't really get this. Template loading in the site editor is based on the `/settings` endpoint, which is required to know the home page ID and posts page ID. So this all doesn't work for non-admins?

Reply:
Non-admins can’t access the site editor, not it’s non-issue for the moment.